scalameta/src/main/scala/scala/meta/Origin.scala

```diff
@@ -35,6 +39,13 @@ object Origin {
     private implicit lazy val thisDialect: Dialect = this.dialect
     val start = input.tokens.apply(startTokenPos).start
     val end = if (endTokenPos != -1) input.tokens.apply(endTokenPos).end else -1
+    private def findLine(x: Int, default: Int) = input match {
+      case Input.None => default
+      case els => els.content.take(x + 1).count(_ == '\n')
+    }
+    val startLine = findLine(start, 0)
+    val endLine = findLine(end, -1)
```

Comment:
A simply way to avoid that would be to have `lazy vals`, since startLine and endLine might not be used much.